marlowe-playground-server/src/Server.hs

```diff
@@ -79,94 +72,28 @@ checkHealth = do
-                                  -- One of these timeout threads is created for every unique analysis request
-                                  void . forkIO $ do
-                                      -- We set the waiting for this unique request as it is the latest
-                                      waiting <- nextRandom
-                                      atomically . modifyTVar registryVar $ startWaiting uuid waiting
-                                      -- Wait for the timeout
-                                      threadDelay (timeoutSeconds * 1000000)
-                                      -- If the user is still waiting for the same request then timeout and tell them
-                                      registry <- readTVarIO registryVar
-                                      if isWaiting uuid waiting registry
-                                        then do
-                                              putStrLn "analysis request timed out"
-                                              sendTextData connection $ encode $ OtherError "Analysis request timed out"
-                                              atomically . modifyTVar registryVar $ finishWaiting uuid
-                                        else pure ()
-
-
-handleNotification :: TVar Registry -> MSRes.Response -> Handler NoContent
-handleNotification registryVar response = liftIO $ do
-    case UUID.fromString $ MSRes.uuid response of
-        Nothing -> putStrLn "Response contains an invalid UUID"
-        Just uuid -> do
-            registry <- readTVarIO registryVar
-            case lookupInRegistry uuid registry of
-                          Nothing -> putStrLn $ "Ignoring response because user " <> show uuid <> " is not in the registry, they've probably disconnected"
-                          Just (_, Nothing) -> putStrLn $ "Ignoring response because user " <> show uuid <> " isn't waiting for one"
-                          Just (connection, _) -> do
-                              atomically . modifyTVar registryVar $ finishWaiting uuid
-                              sendTextData connection $ encode $ CheckForWarningsResult $ MSRes.result response
-    pure NoContent
-
-
 genActusContract :: ContractTerms -> Handler String
 genActusContract terms = pure $ show $ pretty $ genFsContract terms
 
 genActusContractStatic :: ContractTerms -> Handler String
 genActusContractStatic terms = pure $ show $ pretty $ genStaticContract terms
 
+analyseContract :: Text -> Text -> ClientEnv -> CheckForWarnings -> Handler MSRes.Result
+analyseContract apiKey callbackUrl marloweSymbolicClientEnv (CheckForWarnings onlyAssertions contract state) = do
+    let req = MSReq.Request "" (Text.unpack callbackUrl) onlyAssertions contract state
+    res <- liftIO $ runClientM (marloweSymbolicClient (Just "Event") (Just apiKey) req) marloweSymbolicClientEnv
+    case res of
+        Left err                          -> throwError $ err500 {errBody = BSL.pack . show $ err}
```

Comment:
I'm slightly concerned by this. You'll be throwing back things like request headers all the way to the user. That's often a security hole.

Reply:
good spot